src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt

```diff
@@ -50,10 +52,34 @@ class CertificateService(
                     certificateDocumentRequestDto.type.certificateType
                 )
 
+        val existingCertificate = certificateRepository.findByDocumentID(
+            certificateDocumentRequestDto.certificateDocumentID
+        )
+
+        if(existingCertificate != null){
+            throw CertificateAlreadyExists(
+                "Certificate",
+                certificateDocumentRequestDto.certificateDocumentID
+            )
+        }
+
         val entity = certificateMapping.toCertificateDB(certificateDocumentRequestDto, certificateType)
 
         val result = certificateRepository.save(entity)
         return certificateMapping.toCertificateDocumentResponseDto(result)
 
     }
+
+
+    fun retrieveCertificate(cdID: String): ResponseEntity<CertificateDocumentResponseDto> {
+        logger.debug { "Executing retrieveCertificate() with parameters $cdID" }
+
+        val certificate = certificateRepository.findByDocumentID(cdID) ?: throw CertificateTypeNotExists(
```

Comment:
As mentioned above, we need to create new exception for certificate availability check as Certificate and CertificateType are two different entities.